wgengine/magicsock/magicsock.go

```diff
@@ -222,6 +222,59 @@ func (m *peerMap) deleteEndpoint(ep *endpoint) {
 	}
 }
 
+// validate checks m for internal consistency and reports the first error encountered.
```

Comment:
Worth mentioning that this is only used by tests (maybe it should move to a _test.go file?) so it doesn't need to be efficient.

Reply:
Done (both).